src/components/ContentHighlights/HighlightStepper/ContentHighlightStepper.jsx

```diff
@@ -95,103 +106,149 @@ const ContentHighlightStepper = ({ enterpriseId }) => {
       setIsPublishing(false);
     }
   };
+  const closeStepper = () => {
+    open();
+  };
+
+  /* Start beforeunload event */
+  // This section triggers browser response to unsaved items when the stepper modal is open/active
+  useEffect(() => {
+    const preventUnload = () => {
+      /* eslint-disable no-restricted-globals */
+      /* Mandatory requirements to trigger response by browser, event.preventDefault && event.returnValue
+      A return value is required to trigger the browser unsaved data blocking modal response */
+      event.preventDefault();
+      event.returnValue = 'Are you sure? Your data will not be saved.';
+      /* eslint-enable no-restricted-globals */
+    };
+
+    /* Conditional MUST be set on event listener initialization.
+    Failure to provide conditional will trigger browser event on
+    all elements within ContentHighlightRoutes.jsx (essentially all of highlights) */
+    if (isStepperModalOpen) {
+      window.addEventListener('beforeunload', preventUnload);
```

Comment:
nit: use `global` > `window`. we've been bitten elsewhere in the past when React code gets executed at build time, where `window` doesn't exist, but `global` does (i.e., in a Node environment). If this component ever ends up with SSR, it _could_ (depending on whether `window` is called at build time or at runtime) throw an error. I tend to just use `global` to avoid the possibility 🤷‍♂️